initial-ai-src/script.js

```diff
@@ -0,0 +1,587 @@
+    periodSelect.addEventListener("change", () => this.calculate());
+    zoneSelect.addEventListener("change", () => this.calculate());
+    currentDiscountSelect.addEventListener("change", () => this.calculate());
+
+    // Initial calculation
+    this.calculate();
+  }
+
+  getDiscountPercentage(ticketNumber) {
+    if (ticketNumber <= 4) return 0;
+    if (ticketNumber <= 9) return 5;
+    if (ticketNumber <= 14) return 10;
+    if (ticketNumber <= 19) return 15;
+    if (ticketNumber <= 24) return 20;
+    if (ticketNumber <= 29) return 25;
+    if (ticketNumber <= 34) return 30;
+    if (ticketNumber <= 39) return 35;
+    return 40; // Maximum discount
+  }
+
+  getCurrentTicketCount(discountPercentage) {
+    // Convert current discount percentage to approximate ticket count
+    switch (discountPercentage) {
+      case 0:
+        return 0;
+      case 5:
+        return 5;
+      case 10:
+        return 10;
+      case 15:
+        return 15;
+      case 20:
+        return 20;
+      case 25:
+        return 25;
+      case 30:
+        return 30;
+      case 35:
+        return 35;
+      case 40:
+        return 40;
+      default:
+        return 0;
+    }
+  }
+
+  calculateDiscountBreakdown(
+    totalTickets,
+    ticketPrice,
+    currentTicketCount = 0
+  ) {
+    const breakdown = [];
+    let ticketsProcessed = 0;
+
+    while (ticketsProcessed < totalTickets) {
+      const absoluteTicketNumber = currentTicketCount + ticketsProcessed + 1;
+      const discount = this.getDiscountPercentage(absoluteTicketNumber);
+
+      // Find how many tickets at this discount level
+      let ticketsAtThisDiscount = 1;
+      let nextTicketNumber = absoluteTicketNumber + 1;
+
+      while (
+        ticketsProcessed + ticketsAtThisDiscount < totalTickets &&
+        this.getDiscountPercentage(nextTicketNumber) === discount
+      ) {
+        ticketsAtThisDiscount++;
+        nextTicketNumber++;
+      }
+
+      const tierStart = ticketsProcessed + 1;
+      const tierEnd = ticketsProcessed + ticketsAtThisDiscount;
+
+      breakdown.push({
+        range: `${tierStart}-${tierEnd}`,
+        tickets: ticketsAtThisDiscount,
+        discount: discount,
+        pricePerTicket: ticketPrice * (1 - discount / 100),
+        totalCost: ticketsAtThisDiscount * ticketPrice * (1 - discount / 100),
+        absoluteRange: `${absoluteTicketNumber}-${
+          absoluteTicketNumber + ticketsAtThisDiscount - 1
+        }`,
+      });
+
+      ticketsProcessed += ticketsAtThisDiscount;
+    }
+
+    return breakdown;
+  }
+
+  calculate() {
+    const tripsPerWeek =
+      parseInt(document.getElementById("trips-per-week").value) || 0;
+    const period = parseInt(
+      document.getElementById("calculation-period").value
+    );
+    const selectedZone = document.getElementById("zone-selection").value;
+    const currentDiscount = parseInt(
+      document.getElementById("current-discount").value
+    );
```

Comment:
🎨 Why not have error catching like you have for `tripsPerWeek`? Consistency is important 💅